pkg/ring/kv/dynamodb/metrics.go

```diff
@@ -0,0 +1,93 @@
+package dynamodb
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	grpcUtils "github.com/weaveworks/common/grpc"
+	"github.com/weaveworks/common/httpgrpc"
+	"github.com/weaveworks/common/instrument"
+)
+
+type dynamodbInstrumentation struct {
+	kv         dynamodbKV
+	ddbMetrics *dynamodbMetrics
+}
+
+type dynamodbMetrics struct {
+	dynamodbRequestDuration *instrument.HistogramCollector
+	dynamodbUsageMetrics    *prometheus.CounterVec
+}
+
+func newDynamoDbMetrics(registerer prometheus.Registerer) *dynamodbMetrics {
+	dynamodbRequestDurationCollector := instrument.NewHistogramCollector(promauto.With(registerer).NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "dynamodb_kv_request_duration_seconds",
+		Help:    "Time spent on dynamodb requests.",
+		Buckets: prometheus.DefBuckets,
+	}, []string{"operation", "status_code"}))
+
+	dynamodbUsageMetrics := promauto.With(registerer).NewCounterVec(prometheus.CounterOpts{
+		Name: "dynamodb_kv_read_capacity",
```

Comment:
If it is a counter, let's add the counter suffix `_total`.

Reply:
Sure. I didnt see this practices before